scripts/vaccination_alerts/README.md

```diff
@@ -0,0 +1,93 @@
+# Vaccination Alerts
+
+Outline of the workflow:
+
+1. Generate the list of locations that need to be alerted
+2. Generate a list of (user email, location) to be alerted
+3. Send the emails
+
+## 1. generate-vaccine-alerts-info
+
+This script determines which locations have updated vaccination information by comparing the information in Firebase with the information from the CMS.
+
+### Firebase
+
+```yaml
+info:
+  vaccinationInfoUpdates:
+    '11':
+      emailAlertVersion: 0
+    '12':
+      emailAlertVersion: 0
+```
+
+### From the CMS
+
+```yaml
+'11':
+  emailAlertVersion: 3 # Updated
+  content: 'Phase A...'
+'12':
+  emailAlertVersion: 0
+  content: 'Phase 1...'
+'13':
+  emailAlertVersion: 3 # New
+  content: 'Phase 1...'
+```
+
+In this case, for example, we need to send emails for udpates on location "11" and location "13" (location "12" didn't have any updates).
+
+For locations that have updated information, we save a copy of the CMS information in the file `vaccination-alerts.json`. The information in `vaccination-alerts.json` will be used in later phases to determine the list of users to email and to render the emails.
+
+```yaml
+'11':
+  emailAlertVersion: 3
+  content: 'Phase A...'
+'13':
+  emailAlertVersion: 3
+  content: 'Phase 1...'
+```
+
+## 2. generate-users-to-email
+
+Using the locations in `vaccination-alerts.json`, we query email addresses subscribed to these locations from the the Firebase collection `alerts-subscriptions`.
+
+```yaml
+alerts-subscriptions:
+  pablo@covidactnow.org:
+    locations: ['11']
+  chelsi@covidactnow.org:
+    locations: ['11', '13']
+```
+
+Then, we update store the list of locations and email addresses to send in the Firebase collection `vaccination-alerts`, in the document `snapshotId`, so we can track the locations and users that we email each time.
+
+```yaml
+vaccination-alerts:
+  '1500':
+    locations:
+      '11':
+        emails:
+          pablo@covidactnow.org: { sentAt: null }
+          chelsi@covidactnow.org: { sentAt: null }
+      '13':
+        emails:
+          chelsi@covidactnow.org: { sentAt: null }
```

Comment:
Hrm...  Keying off of the `snapshotId` made sense for risk alerts, since it uniquely identifies the data that we are alerting off of...  and in particular, this structure allowed us to make the alert sending process resumable / idempotent so that if the alert sending failed halfway through for snapshot 1500, you could just run it again and it would ignore everybody we already emailed (i.e. non-null `sentAt`) and resume sending to everybody else (with null `sentAt`).

To get a similar behavior, I think maybe we'd want a structure something like:
```yaml
vaccination-alerts:
  '11':
    emailAlertVersions:
      '0':
        emails:
          pablo@covidactnow.org: { sentAt: 'yyyy/mm/dd ...' }
          chelsi@covidactnow.org: { sentAt: 'yyyy/mm/dd ...' }
      '3':
        emails:
          pablo@covidactnow.org: { sentAt: 'yyyy/mm/dd ...' }
          chelsi@covidactnow.org: { sentAt: null }
```

This would indicate that we previously successfully sent alerts for location 11 for version 0, but sending alerts for version 3 failed partway through and so you got the email but chelsi didn't.  When we re-ran the script, it should automatically send it to Chelsi but not you.

Feel free to ping me if this doesn't make sense or you want to talk through in more detail.

Reply:
alternatively, we could have:

```
vaccination-alerts:
  '11':
        emails:
          pablo@covidactnow.org: { sentAt: 'yyyy/mm/dd ...', lastAlertVersion: '0' }
          chelsi@covidactnow.org: { sentAt: 'yyyy/mm/dd ...', lastAlertVersion: '3' }
```
or

```
vaccination-alerts:
  '11':
        emails:
          pablo@covidactnow.org: 
            alertVersion:
              0: { sentAt: 'yyyy/mm/dd ...' }
              3: { sentAt: 'yyyy/mm/dd ...' }
          chelsi@covidactnow.org: 
            alertVersion:
              0: { sentAt: 'yyyy/mm/dd ...' }
```

Reply:
I'm going with Michael's recommendation, it feels easier to query if we want stats on the emails